tests/test_cli/test_generate_all_protocols.py

```diff
@@ -0,0 +1,267 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2022 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Test `generate-all-protocols` command."""
+
+import logging
+import pprint
+import shutil
+import subprocess
+from pathlib import Path
+from typing import Any
+from unittest import mock
+
+import pytest
+
+from aea.configurations.constants import (
+    DEFAULT_PROTOCOL_CONFIG_FILE,
+    DEFAULT_README_FILE,
+)
+from aea.exceptions import enforce
+from aea.manager.helpers import AEAProject
+from aea.test_tools.exceptions import AEATestingException
+from aea.test_tools.test_cases import AEATestCaseMany
+
+
+def _run_cli_patch(*args: Any, **kwargs: Any) -> None:
+
+    if "isort" in args or "black" in args:
+        return
+
+    print(f"Calling command {args} with kwargs {kwargs}")
+    return_code = subprocess.check_call(args, **kwargs)  # nosec
+    enforce(
+        return_code == 0,
+        f"Return code of {pprint.pformat(args)} is {return_code} != 0.",
+    )
+
+
+class TestGenerateAllProtcols(AEATestCaseMany):
+    """Test generate all protocols."""
+
+    use_packages_dir: bool = True
+    find_packages_patch: Any
+    package_path: Path
+
+    @classmethod
+    def setup_class(cls) -> None:
+        """Setup test class."""
+        super().setup_class()
+
+        cls.package_path = cls.t / "packages" / "fetchai" / "protocols" / "fipa"
```

Comment:
Define on top as a constant, not in the function. Better yet, import it
```
from packages.fetchai.protocols import fipa
...
cls.package_path = Path(fipa.__file__).parent
```

Reply:
We already know the path for the package so there is no point in import the package and using it to define the path. Also no point in defining a top level constant since this refers to a temporary directory which changes everytime the test runs.